### PR TITLE
Frame buffer API for user space and lvgl port

### PR DESCRIPTION
**Pull request by alexandruradovici**
This pull request:
1. adds the frame buffer API for https://github.com/tock/tock/pull/1837.
2. adds a simple driver for lvgl (https://github.com/littlevgl/lvgl)

### CODE REVIEW

**Comment by alexandruradovici** [comment]
The frame buffer HIL has been merged in Tock as the screen HIL. I renamed the userland library to screen.

File: lvgl/Makefile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+$(LIBNAME)_DIR := $(TOCK_USERLAND_BASE_DIR)/$(LIBNAME)
+
+
+DEPPATH += --dep-path $(LVGL_DIR)
+VPATH += :$(LVGL_DIR)
+CSRCS += 
+
+override CFLAGS += -Wno-error=unused-macros -Wno-error=sign-compare -Wno-error=float-equal\
+  -Wno-error=unused-parameter -Wno-error=unused-variable\
+  -Wno-error=shadow -Wno-error=type-limits -Wno-error=overflow
```
Comment (ppannuto) [inline lvgl/Makefile:12]: These are disappointing, but not really our problem to fix.

File: lvgl/Makefile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+LVGL_DIR = $(TOCK_USERLAND_BASE_DIR)/lvgl
+
+include $(LVGL_DIR)/lvgl/lvgl.mk
```
Comment (ppannuto) [inline lvgl/Makefile:16]: This is always scary, but glad it works!

File: libtock/st7735.h [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define DRIVER_NUM_ST7735 0x80006
+
+#define	ST7735_COLOR_BLACK   0x0000
+#define	ST7735_COLOR_BLUE    0x001F
+#define	ST7735_COLOR_RED     0xF800
+#define	ST7735_COLOR_GREEN   0x07E0
```
Comment (ppannuto) [inline libtock/st7735.h:20]:
```suggestion
#define ST7735_COLOR_BLACK   0x0000
#define ST7735_COLOR_BLUE    0x001F
#define ST7735_COLOR_RED     0xF800
#define ST7735_COLOR_GREEN   0x07E0
```

File: libtock/st7735.h [hunk 4]
```diff
@@ -0,0 +1,31 @@
+/*ST7735 LCD screen
+*
+* <https://www.pololu.com/file/0J563/L3GD20.pdf>
```
Comment (ppannuto) [inline libtock/st7735.h:3]: Wrong link?

File: libtock/screen.h [hunk 5]
```diff
@@ -0,0 +1,66 @@
+/*L3GD20 3 axis gyroscope and temperature sensor
+*
+* <https://www.pololu.com/file/0J563/L3GD20.pdf>
+*
+*/
```
Comment (ppannuto) [inline libtock/screen.h:5]: This should probably be something more about the general screen interface?

File: libtock/screen.c [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+}
+
+int screen_get_supported_resolutions (void) {
+  ScreenReturn fbr;
+  fbr.data1 = 0;
+  fbr.done  = false;
+  screen_subscribe (screen_callback, &fbr);
+  fbr.error = screen_command (11, 0, 0);
+  if (fbr.error == TOCK_SUCCESS) yield_for (&fbr.done);
+  return fbr.data1;
```
Comment (ppannuto) [inline libtock/screen.c:45]: Should this return `fbr.error` in the case that the original command error'd? (Here, and later throughout this file)

File: examples/lvgl/main.c [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+    lv_obj_align(label1, NULL, LV_ALIGN_CENTER, 0, 0);
+
+    /* main loop */
+    while (1)
+    {
+      delay_ms (5);
+      lvgl_driver_event (5);
+    }
+  } else {
+    printf ("lvgl init error\n");
```
Comment (ppannuto) [inline examples/lvgl/main.c:31]: Might be nice to capture and print the error code here

File: libtock/screen.h (hunk 5 above)
Comment (alexandruradovici) [reply libtock/screen.h:5]: Yes, sorry for this

File: lvgl/Makefile (hunk 1 above)
Comment (alexandruradovici) [reply lvgl/Makefile:12]: I added them when I thought the check will fail due to library warnings, I think we can remove it.

File: libtock/st7735.h (hunk 4 above)
Comment (alexandruradovici) [reply libtock/st7735.h:3]: Yes

File: libtock/screen.c (hunk 6 above)
Comment (alexandruradovici) [reply libtock/screen.c:45]: This is correct, as the function returns the number of available resolutions.

File: examples/lvgl/main.c (hunk 7 above)
Comment (alexandruradovici) [reply examples/lvgl/main.c]: Done

**Pull request merged**